filament/src/details/SkinningBuffer.cpp

```diff
@@ -188,10 +188,16 @@ inline size_t getSkinningBufferSize(size_t pairCount) noexcept {
 UTILS_NOINLINE
 void updateDataAt(backend::DriverApi& driver,
         Handle<HwTexture> handle, PixelDataFormat format, PixelDataType type,
-        const char* out, size_t elementSize,
-        size_t count, size_t size) {
+        const utils::FixedCapacityVector<math::float2>& pairs,
+        size_t count) {
 
-    size_t const textureWidth   = getSkinningBufferWidth( count);//size);
+    size_t elementSize = sizeof(float2);
+    size_t size = getSkinningBufferSize(count);
+    auto* out = (float2*) malloc(size);
+    std::transform(pairs.begin(), pairs.begin() + count, out,
+            [](const float2& p) { return float2(p); });
```

Comment:
isn't that basically just `memcpy` ?

Reply:
You're right, it's just memcpy now. I repaired. Originally the texture format could have been more general. Thank you